mlrun/runtimes/sparkjob.py

```diff
@@ -249,36 +248,42 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
             update_in(job, "spec.sparkVersion", self.spec.spark_version)
 
         if self.spec.restart_policy:
-            update_in(job, "spec.restartPolicy.type", self.spec.restart_policy["type"])
-            update_in(
+            verify_and_update_in(job, "spec.restartPolicy.type", self.spec.restart_policy["type"], str)
+            verify_and_update_in(
                 job,
                 "spec.restartPolicy.onFailureRetries",
                 self.spec.restart_policy["retries"],
+                int,
             )
-            update_in(
+            verify_and_update_in(
                 job,
                 "spec.restartPolicy.onFailureRetryInterval",
                 self.spec.restart_policy["retry_interval"],
+                int,
             )
-            update_in(
+            verify_and_update_in(
                 job,
                 "spec.restartPolicy.onSubmissionFailureRetries",
                 self.spec.restart_policy["submission_retries"],
+                int,
             )
-            update_in(
+            verify_and_update_in(
                 job,
                 "spec.restartPolicy.onSubmissionFailureRetryInterval",
                 self.spec.restart_policy["submission_retry_interval"],
+                int,
             )
 
         update_in(job, "metadata", meta.to_dict())
         update_in(job, "spec.driver.labels", pod_labels)
         update_in(job, "spec.executor.labels", pod_labels)
-        update_in(job, "spec.executor.instances", self.spec.replicas or 1)
+        verify_and_update_in(job, "spec.executor.instances", self.spec.replicas or 1, int)
         update_in(job, "spec.nodeSelector", self.spec.node_selector or {})
 
         if (not self.spec.image) and self._default_image:
             self.spec.image = self._default_image
+
+        verify_field_of_type("spec.image", self.spec.image, str)
```

Comment:
Why not verify_and_update? Why do you check self.spec.image and not the actual value we put there?

Reply:
the update_in is receiving spec.full_image_path() as parameter and not the original value the user has entered

Reply:
It doesn't matter. We just validate the final result injected to the SparkApplication CRD